js/battledata.js

```diff
@@ -1024,6 +1025,18 @@ var Tools = {
 				}
 				template.formeid = formeid;
 			}
+			if (template.otherForms) {
+				var form = '', formid = '';
+				if (template.otherForms.indexOf(speciesid) >= 0) {
+					form = speciesid.slice(template.baseSpecies.length);
+					formid = '-' + form;
+					form = form[0].toUpperCase() + form.slice(1);
+				}
+				template.form = form;
```

Comment:
Ideally, you're not overwriting data in `BattlePokedex`.

You might want to make another table, and clone templates into that table (shallow-clones are fine here).